### PR TITLE
Improve VSCode instructions in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by sffc**
Follow-up to #4486

### CODE REVIEW

**Comment by sffc** [comment]
Which is better and which is worse:

image

image

**Comment by robertbastian** [comment]
Put the comment outside of the code block? People will copy paste it, it should be valid JSON.

**Comment by sffc** [comment]
VSCode allows comments in the JSON.

**Comment by robertbastian** [comment]
It also makes it seem like you have to update the path, but you don't

**Comment by sffc** [comment]
TIL that one of the syntaxes supported here is "JSON with Comments", alias `jsonc`

https://github.com/github-linguist/linguist/blob/559a6426942abcae16b6d6b328147476432bf6cb/lib/linguist/languages.yml#L3195

But it appears to use the same colors as `javascript`

**Comment by robertbastian** [comment]
Yes paths are complicated, but this one isn't. The location of the stub data relative to the include statement is fixed. Usually when there's a comment like this next to a configuration that I'm copy-pasting, I'm supposed to change that; that's not the case here.

**Review by Manishearth — DISMISSED** [review]
ambivalent on using jsonc

**Comment by sffc** [comment]
Thought: Do you think we should make VSCode use a different Cargo profile? If you run `cargo test --all-features` on the CLI and then VSCode, I think it causes compilation with different flags to go into the same target folder.

**Comment by robertbastian** [comment]
I think that's a setting everyone can decide for themselves.

**Review by robertbastian — COMMENTED** [review]
.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -30,18 +30,19 @@ ICU4X can be edited using any text editor capable of editing Rust code.
...
-To build all code paths and improve build times in VSCode, we recommend the following settings. (Note: The second setting causes VSCode to build ICU4X with only the `und` locale, which reduces build times but also makes some tests fail; to run them normally, run `cargo test --all-features` on the command line.) To add these settings, choose "Preferences: Open Workspace Settings (JSON)" from the command palette (Ctrl+Shift+P):
+To build all code paths and improve build times in VSCode, we recommend the following settings. To add them, choose "Preferences: Open Workspace Settings (JSON)" from the command palette (Ctrl+Shift+P):
 
-```javascript
+```json
 "settings": {
 	"rust-analyzer.cargo.features": "all",
 	"rust-analyzer.cargo.extraEnv": {
-		// Path relative to `provider/baked/*/src/lib.rs`
 		"ICU4X_DATA_DIR": "../../../../provider/datagen/tests/data/baked"
```
Comment (robertbastian) [inline CONTRIBUTING.md:39]:
```suggestion
		"ICU4X_DATA_DIR": "../../../datagen/tests/data/baked"
```

**Review by robertbastian — DISMISSED** [review]

**Review by robertbastian — APPROVED** [review]

**Pull request merged**